react-playground/src/components/MenuItem/MenuItem.tsx

```diff
@@ -0,0 +1,28 @@
+import { FC } from "react";
+import { Link } from "react-router-dom";
+
+export type PropMenuItem = {
+  name: String;
+  route: String;
+};
+interface MenuItemProps {
+  className?: string;
+  item: PropMenuItem;
+  onItemClicked: (item: PropMenuItem) => void;
```

Comment:
So actually, since all you need is for <Link> to trigger a close of the menu, I'd maybe recommend passing the prop `closeMenu` as a callback function instead. Below, use:
`<Link (...your other props here) onClick={closeMenu}>`

Reply:
Getting the following error when I use an `onClick` on Link:

ERROR in src/components/MenuItem/MenuItem.tsx:22:9

TS2322: Type 'void' is not assignable to type 'MouseEventHandler<HTMLAnchorElement> | undefined'.
    20 |         to={`${props.item.route}`}
    21 |         style={{ color: "inherit", textDecoration: "inherit" }}
  > 22 |         onClick={props.closeMenu()}
       |         ^^^^^^^
    23 |       >
    24 |         {props.item.name}
    25 |       </Link>

I think it's because I have an eventListener for clicks and that overriding this function

Reply:
I fixed this

Reply:
Nice! I didn't have a good answer to give without looking but I wondered if onClick={() => closeMenu()) would work or if casting as MouseEventHandler was possible